lighthouse-core/gather/computed/trace-of-tab.js

```diff
@@ -32,6 +32,35 @@ class TraceOfTab extends ComputedArtifact {
     return 'TraceOfTab';
   }
 
+  /**
+   * @param {LH.TraceEvent[]} traceEvents
+   * @param {(e: LH.TraceEvent) => boolean} filter
+   */
+  static filteredStableSort(traceEvents, filter) {
+    // create an array of the indices that we want to keep
+    const indices = [];
+    for (let srcIndex = 0; srcIndex < traceEvents.length; srcIndex++) {
+      if (filter(traceEvents[srcIndex])) {
```

Comment:
it was 3-pronged

1) 
```js
TraceOfTab.stableSort(
  traceEvents.filter()
)
```
was gross
2) avoid another copy of the array since we're introducing a new one here and there were recent issues with large traces hurting people
3) filter inline here was ever so mildly faster on big traces

none of them are particularly strong reasons, so if it's rubbing the wrong way I can go back

Reply:
I'm seeing the perf benefit of the inline filter to be substantial. I think all numbers are under 50ms even for big traces, but it's worth like a 35% speedup.``

Reply:
FWIW https://jsperf.com/inline-filter-vs-filter

Reply:
ha, if you're fine on a style level with combining the two steps, @paulirish, I can be fine with it too. Testing it now I see ~5% improvement over all of `_compute()` for a 31MB verge trace (~138ms -> ~131). I was just questioning this part from a code-semantics perspective.

Reply:
(it's likely not better than 5% because the `filteredStableSort` call goes polymorphic with the second call to it in `_compute()`, while AIUI as of Node 10 `[].filter()` specializes at the callsite, not within.

This may get worse as more calls are made, e.g. if someone is running the node module in a loop. You can kind of simulate this by disabling the computed artifact cache and letting `trace-of-tab` run as many times as requested in a `-G` run (110 times!). In that case the new method is a little slower (by ~5.3±4ms)

![bench-plot](https://user-images.githubusercontent.com/316891/41797233-2430b202-761e-11e8-82d2-a6d148de2b88.png)

All of this is dependent on inlining decisions, though, and how lighthouse itself is run, so perf seems fine either way)

Reply:
hot chart!

(yeah the code style doesn't bother me too much)